nvflare/app_common/pt/pt_scaffold.py

```diff
@@ -44,81 +44,77 @@
 from torch.optim import Optimizer
 
 
-class PTScaffold(object):
-    """Learner Spec to be used with SCAFFOLD components.
+def get_lr_values(optimizer: Optimizer):
+    """
+    This function is used to get the learning rates of the optimizer.
+    """
+    return [group["lr"] for group in optimizer.state_dict()["param_groups"]]
+
+
+class PTScaffoldHelper(object):
+    """Helper to be used with SCAFFOLD components.
     Implements the functions used for the algorithm proposed in
     Karimireddy et al. "SCAFFOLD: Stochastic Controlled Averaging for Federated Learning"
     (https://arxiv.org/abs/1910.06378) using PyTorch.
     SCAFFOLD-related functions are based on https://github.com/Xtra-Computing/NIID-Bench.
     See also Li et al. "Federated Learning on Non-IID Data Silos: An Experimental Study"
     (https://arxiv.org/abs/2102.02079).
-
-    Note: functions inherited from `Learner` are abstract and need to be implemented by the app.
     """
 
     def __init__(self):
-        # Model being optimized in SCAFFOLD
-        self.model = None  # should be provided by the Learner implementation
-
         # SCAFFOLD control terms
         self.cnt = 0
-        self.curr_lr = None
         self.c_global = None
         self.c_local = None
         self.c_delta_para = None
 
-    def scaffold_init(self):
-        # Should be called after `model` in Learner implementation is initialized.
-        if not self.model:
-            raise ValueError("SCAFFOLD model is not initialized!")
-
+    def init(self, model):
         # create models for SCAFFOLD correction terms
-        self.c_global = copy.deepcopy(self.model)
-        self.c_local = copy.deepcopy(self.model)
+        self.c_global = copy.deepcopy(model)
+        self.c_local = copy.deepcopy(model)
         # Initialize correction term with zeros
-        c_init_para = self.model.state_dict()
+        c_init_para = model.state_dict()
         for k in c_init_para.keys():
             c_init_para[k] = torch.zeros_like(c_init_para[k])
         self.c_global.load_state_dict(c_init_para)
         self.c_local.load_state_dict(c_init_para)
 
-    def scaffold_get_params(self):
+    def get_params(self):
         self.cnt = 0
         # Adapted from https://github.com/Xtra-Computing/NIID-Bench/blob/main/experiments.py#L371
         c_global_para = self.c_global.state_dict()
         c_local_para = self.c_local.state_dict()
         return c_global_para, c_local_para
 
-    def scaffold_model_update(self, c_global_para, c_local_para):
+    def model_update(self, model, curr_lr, c_global_para, c_local_para):
```

Comment:
just wonder the arg order `model_update(self, model, curr_lr, c_global_para, c_local_para)`

VS 

`terms_update(self, model, model_global, c_global_para, c_local_para, curr_lr)`

They both have model, curr_lr, c_global_para, c_local_para, do you think it would be good if they have similar order in both functions.

Reply:
Good point. I made it consitent.